src/main/java/panicathe/autumnfintech/service/AuthService.java

```diff
@@ -0,0 +1,71 @@
+package panicathe.autumnfintech.service;
+
+import jakarta.persistence.EntityNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import panicathe.autumnfintech.dto.user.LoginRequestDto;
+import panicathe.autumnfintech.dto.user.UserDto;
+import panicathe.autumnfintech.entity.User;
+
+import panicathe.autumnfintech.jwt.JwtProvider;
+import panicathe.autumnfintech.repository.UserRepository;
+
+import java.util.Objects;
+
+@Service
+@RequiredArgsConstructor
+public class AuthService {
+
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+    private final JwtProvider jwtProvider;
+
+
+    @Transactional
+    public void register(UserDto userDto) {
+        if (userRepository.existsByEmail(userDto.getEmail())) {
+            throw new IllegalArgumentException("Email already exists.");
+        }
+        if(userRepository.existsByUsername(userDto.getUsername()))
+            throw new IllegalArgumentException("Username already exists.");

```

Comment:
```suggestion
        if (userRepository.existsByUsername(userDto.getUsername())) {
            throw new IllegalArgumentException("Username already exists.");
        }
```
if 문 작성시 coding convention 은 위와 같으니 지킬 수 있도록 노력해보시면 좋겠네요 ^^;
https://naver.github.io/hackday-conventions-java/